FWIW, I've been working on a new feature for my bot to try to build subsets of big PRs and this one came at the perfect time for local testing.

It looks like `pyopnsense` might be failing here with the following message from my testing
```
==================================== ERRORS ====================================
_______________ ERROR collecting pyopnsense/tests/test_client.py _______________
ImportError while importing test module '/build/pyopnsense-0.3.0/pyopnsense/tests/test_client.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/3qsj5r2w2z0dqj7ri70jifaq5ks3rc42-python3-3.9.1/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
pyopnsense/tests/test_client.py:25: in <module>
    from pyopnsense.tests import base
pyopnsense/tests/base.py:18: in <module>
    import fixtures
/nix/store/hd7dcbrzn8vxrx1mmya3yl7wh25awy00-python3.9-fixtures-3.0.0/lib/python3.9/site-packages/fixtures/__init__.py:81: in <module>
    from fixtures.fixture import (
/nix/store/hd7dcbrzn8vxrx1mmya3yl7wh25awy00-python3.9-fixtures-3.0.0/lib/python3.9/site-packages/fixtures/fixture.py:29: in <module>
    from testtools.compat import (
E   ModuleNotFoundError: No module named 'testtools'
=========================== short test summary info ============================
```

--- a/pkgs/development/python-modules/fixtures/default.nix
+++ b/pkgs/development/python-modules/fixtures/default.nix
@@ -1,11 +1,12 @@
 { lib
 , buildPythonPackage
 , fetchPypi
+, fetchpatch
+, pythonAtLeast
 , pbr
 , testtools
 , mock
 , python
-, isPy39
 }:
 
 buildPythonPackage rec {
@@ -16,6 +17,15 @@ buildPythonPackage rec {
     inherit pname version;
     sha256 = "fcf0d60234f1544da717a9738325812de1f42c2fa085e2d9252d8fff5712b2ef";
   };
+
+  patches = lib.optional (pythonAtLeast "3.9") [
+    # drop tests that try to monkeypatch a classmethod, which fails on python3.9
+    # https://github.com/testing-cabal/fixtures/issues/44
+    (fetchpatch {
+       url = "https://salsa.debian.org/openstack-team/python/python-fixtures/-/raw/debian/victoria/debian/patches/remove-broken-monkey-patch-test.patch";
+       sha256 = "1s3hg2zmqc4shmnf90kscphzj5qlqpxghzw2a59p8f88zrbsj97r";
+    })
+  ];
 
   nativeBuildInputs = [
     pbr
@@ -37,6 +47,5 @@ buildPythonPackage rec {
     description = "Reusable state for writing clean tests and more";
     homepage = "https://pypi.python.org/pypi/fixtures";
     license = lib.licenses.asl20;
-    broken = isPy39; # see https://github.com/testing-cabal/fixtures/issues/44
   };
 }

--- a/pkgs/development/python-modules/fixtures/default.nix
+++ b/pkgs/development/python-modules/fixtures/default.nix
@@ -17,7 +17,17 @@ buildPythonPackage rec {
     sha256 = "fcf0d60234f1544da717a9738325812de1f42c2fa085e2d9252d8fff5712b2ef";
   };
 
-  propagatedBuildInputs = [ pbr testtools mock ];
+  nativeBuildInputs = [
+    pbr
+  ];
+
+  propagatedBuildInputs = [
+    testtools
+  ];
+
+  checkInputs = [
+    mock
+  ];
 
   checkPhase = ''
     ${python.interpreter} -m testtools.run fixtures.test_suite

